test/e2e/node_feature_discovery.go

```diff
@@ -260,27 +434,153 @@ var _ = framework.KubeDescribe("Node Feature Discovery", func() {
+
+				fakeFeatureLabels := map[string]string{
+					master.LabelNs + "fake-fakefeature1": "true",
+					master.LabelNs + "fake-fakefeature2": "true",
+					master.LabelNs + "fake-fakefeature3": "true",
+				}
+
+				// Remove pre-existing stale annotations and labels
+				cleanupNode(f.ClientSet)
+
+				// Launch nfd-worker
+				By("Creating a nfd worker pod")
+				image := fmt.Sprintf("%s:%s", *dockerRepo, *dockerTag)
+				workerPod := nfdWorkerPod(image, []string{"--oneshot", "--sources=fake"})
+				workerPod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(workerPod)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Waiting for the nfd-worker pod to succeed")
+				Expect(e2epod.WaitForPodSuccessInNamespace(f.ClientSet, workerPod.ObjectMeta.Name, f.Namespace.Name)).NotTo(HaveOccurred())
+				workerPod, err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(workerPod.ObjectMeta.Name, metav1.GetOptions{})
+
+				By(fmt.Sprintf("Making sure '%s' was decorated with the fake feature labels", workerPod.Spec.NodeName))
+				node, err := f.ClientSet.CoreV1().Nodes().Get(workerPod.Spec.NodeName, metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				for k, v := range fakeFeatureLabels {
+					Expect(node.Labels[k]).To(Equal(v))
+				}
+
+				// Check that there are no unexpected NFD labels
+				for k := range node.Labels {
+					if strings.HasPrefix(k, master.LabelNs) {
+						Expect(fakeFeatureLabels).Should(HaveKey(k))
+					}
+				}
+
+				By("Deleting the node-feature-discovery worker pod")
+				err = f.ClientSet.CoreV1().Pods(f.Namespace.Name).Delete(workerPod.ObjectMeta.Name, &metav1.DeleteOptions{})
+
+				cleanupNode(f.ClientSet)
+			})
+		})
 
-			By("Removing the fake feature labels advertised by the node-feature-discovery pod")
-			for key := range fakeFeatureLabels {
-				framework.RemoveLabelOffNode(f.ClientSet, workerPod.Spec.NodeName, key)
-			}
+		//
+		// More comprehensive test when --e2e-node-config is enabled
+		//
+		Context("and nfd-workers as a daemonset with default sources enabled", func() {
+			It("the node labels listed in the e2e config should be present", func() {
+				readConfig()
+				if conf == nil {
+					Skip("no e2e-config was specified")
+				}
+				if conf.DefaultFeatures == nil {
+					Skip("no 'defaultFeatures' specified in e2e-config")
+				}
+				fConf := conf.DefaultFeatures
+
+				// Remove pre-existing stale annotations and labels
+				cleanupNode(f.ClientSet)
+
+				By("Creating nfd-worker daemonset")
+				workerDS := nfdWorkerDaemonSet(fmt.Sprintf("%s:%s", *dockerRepo, *dockerTag), []string{})
+				workerDS, err := f.ClientSet.AppsV1().DaemonSets(f.Namespace.Name).Create(workerDS)
+				Expect(err).NotTo(HaveOccurred())
+
+				By("Waiting for daemonset pods to be ready")
+				Expect(e2epod.WaitForPodsReady(f.ClientSet, f.Namespace.Name, workerDS.Spec.Template.Labels["name"], 5)).NotTo(HaveOccurred())
+
+				//By("Sleeping....")
+				//time.Sleep(120 * time.Second)
```

Comment:
Do we need this or should we remote it?

Reply:
Woops 😬 Good catch, good that you're really reading the code. Fixed

Reply:
I.e. removed